### ISSUE TITLE
Consider incorporating this into concourse-boshrelease?

### ISSUE THREAD

**Issue by aegershman** [cirocosta/flight_recorder#6]
Hey there, forgive the informality of this issue, but wanted to suggest `flight_recorder` be packaged in such a way that it could be optionally consumed as a job in the [`concourse-bosh-deployment`](https://github.com/concourse/concourse-bosh-deployment) (or perhaps [`prometheus-boshrelease` directly, similar to how other exporters are packaged in there?](https://github.com/bosh-prometheus/prometheus-boshrelease)) so that interesting data like number of pipelines, teams, resources, etc. could be scraped.

Curious if you have any thoughts on this? Not urgent. Feel free to close whenever, thanks for your time 👍

**Comment by cirocosta** [comment]
Heey Aaron!

soorrry for the super long delay!

<br />

I was taking a look at the Prometheus BOSH release that you mentioned, and it
*seems* like it'd be possible to use the PostgreSQL exporter
(https://github.com/wrouesnel/postgres_exporter) that's bundled in it directly
to make the same queries that `flight_recorder` does :thinking: **Have you checked
out if it's possible to put some custom queries there?**

If so, I imagine it'd be quite simple to copy the queries that `flight_recorder`
does:

https://github.com/cirocosta/flight_recorder/blob/77358155c32ef1a5dfd77ccf12cdb7cf1e47acd9/db/containers.go#L8-L20

That approach similar to how other folks that use datadog have been doing to get
answers from the database directly using the `dd-agent` bosh release (leveraging
the `postgres` integration:
https://docs.datadoghq.com/integrations/faq/postgres-custom-metric-collection-explained/)

```yaml
  - name: dd-agent
    properties:
      dd:
        api_key: bla
        bosh_tags_prefix: myconcourse_
        integrations:
          postgres:
            instances:
            - custom_queries:
              - columns:
                - name: builds_pending
                  type: gauge
                metric_prefix: concourse
                query: |
                  SELECT count(1)
                  FROM builds b, pipelines p
                  WHERE b.status='pending' and p.id=b.pipeline_id and p.paused=false;
```

(I realize that's pretty much me making excuses to not go through creating a
BOSH release :sweat_smile: - it's been a while since I've done it)

I know it's not suuper hard (https://bosh.io/docs/create-release/ and
https://www.youtube.com/watch?v=pXbHepMNhZY are great resources), but I'm just
afraid having to maintain it is harder than just building a container image
:grimacing:

---

for curiosity sake, If some of these metrics were to be natively exported by
Concourse somehow, which would you need? wondering if we should just bake some
of these in somehow :thinking: I've been paying close attention to the development
of [opentelemetry] so that we can soon simplify a lot how we deal with metrics in
the codebase, making easier to have other components other than ATC emitting
metrics, so I can totally see in a soon-to-come future having a component that'd deal
with these very valuable higher-level metrics :thinking: 

[opentelemetry]: https://opentelemetry.io/

thanks!

**Comment by aegershman** [comment]
That's fair enough, I haven't tried with the bosh-prometheus postgres exporter, someday when there's time I'll give that a shot.

Honestly, what I crave are the kinds of metrics which are "demonstration of value"-type metrics, such as number of pipelines total, pipelines by team, number of teams, number of users, average number of job failures, average job completion time length, number of concurrent job builds, total number of resources & check-length time, concurrent checks per resource (how many `git` resources are pounding github at once? etc.), [slis regarding `sync`s and `logins` and `sets`, etc.](https://metrics-hush-house.concourse-ci.org/d/6Tsoq67Wz/sli?orgId=1&refresh=1m), maybe [**user actions**](https://metrics-hush-house.concourse-ci.org/d/1mUBdWWWk/stackdriver?orgId=1&refresh=1m), etc., etc., etc.

for example, if the tools were available to (generally) demonstrate "average job time per team" or "resource usage per team", that definitely help determine where isolated workers could go, or demonstrate why certain business units should be paying more 😉 

Basically anything that can be used to demonstrate the value of concourse to consumers and higher-ups (to justify costs, etc.) are incredibly valuable.

Thanks a lot for your time; I think we can close this out 👍
